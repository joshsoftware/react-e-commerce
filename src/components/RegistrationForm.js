import React from 'react';
import { Form } from 'reactstrap';
import { Link } from 'react-router-dom';
import ButtonWrapper from './ButtonWrapper';
import FormEmailField from './FormEmailField';
import FormPasswordField from './FormPasswordField';
import FormTextField from './FormTextField';
import PropTypes from 'prop-types';
import FormDropdownField from './FormDropdownField';
import alertReducer from '../reducers/alertReducer';
import { useDispatch } from 'react-redux';
import { alertMessage } from '../actions/alertActions';

const RegistrationForm = ({ validateData, dispatch, formState }) => {
  const {
    firstname,
    lastname,
    email,
    password,
    country,
    state,
    city,
    address,
    firstnameError,
    lastnameError,
    emailError,
    passwordError,
    addressError,
    isLoading
  } = formState;
  const alertDispatch = useDispatch(alertReducer);
  alertDispatch(alertMessage({ alert: true, alertText: 'Successfully Registered' }));
  return (
    <>
      <h3>Register</h3>
      <hr />
      <Form
        onSubmit={(e) => {
          e.preventDefault();
          validateData();
        }}>
        <FormEmailField
          email={email}
          emailError={emailError}
          dispatch={dispatch}
          isRequired={true}
        />
        <FormPasswordField
          password={password}
          passwordError={passwordError}
          dispatch={dispatch}
          isRequired={true}
        />
        <FormTextField
          firstname={firstname}
          lastname={lastname}
          address={address}
          firstnameError={firstnameError}
          lastnameError={lastnameError}
          addressError={addressError}
          dispatch={dispatch}
          isRequired={true}
        />
        <FormDropdownField country={country} state={state} city={city} dispatch={dispatch} />
        <br />
        <ButtonWrapper
          buttonText={'Register'}
          onSubmit={(e) => {
            e.preventDefault();
            console.log('registration');
            validateData();
          }}
          disabled={isLoading}
        />
      </Form>
      <br />
      <h6>
        Already registered?
        <span className="input-group-btn">
          <Link to="/login"> Click to login</Link>
        </span>
      </h6>
    </>
  );
};

export default RegistrationForm;

RegistrationForm.propTypes = {
  validateData: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
  formState: PropTypes.object.isRequired
};
