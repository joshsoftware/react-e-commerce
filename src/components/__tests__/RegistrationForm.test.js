import { render, cleanup } from '@testing-library/react';
import React from 'react';
import RegistrationForm from '../RegistrationForm';

afterEach(cleanup);

describe('RegistrationForm Component', () => {
  it('Must return RegistrationForm Component', () => {
    const { asFragment } = render(<RegistrationForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
