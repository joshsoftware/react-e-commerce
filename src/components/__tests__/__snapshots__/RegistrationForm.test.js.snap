// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component Must return RegistrationForm Component 1`] = `
<DocumentFragment>
  <h3>
    Register
  </h3>
  <hr />
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleEmail"
      >
        Email
      </label>
      <input
        class="form-control"
        name="email"
        placeholder="example@company.com"
        type="email"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="examplePassword"
      >
        Password
      </label>
      <input
        class="form-control"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleFirstName"
      >
        First Name
      </label>
      <input
        class="form-control"
        name="firstname"
        placeholder="John"
        type="firstname"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleLastName"
      >
        Last Name
      </label>
      <input
        class="form-control"
        name="lastname"
        placeholder="Doe"
        type="lastname"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleCountry"
      >
        Country
      </label>
      <input
        class="form-control"
        name="country"
        placeholder=""
        type="country"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleState"
      >
        State
      </label>
      <input
        class="form-control"
        name="state"
        placeholder=""
        type="state"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleCity"
      >
        City
      </label>
      <input
        class="form-control"
        name="city"
        placeholder=""
        type="city"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=""
        for="exampleAddress"
      >
        Address
      </label>
      <input
        class="form-control"
        name="address"
        placeholder=""
        type="address"
        value=""
      />
      <div
        class="invalid-feedback"
      />
    </div>
    <button
      class="btn btn-secondary"
    >
      Submit
    </button>
  </form>
  <h6>
    <br />
    Already registered? login
  </h6>
</DocumentFragment>
`;
